tests/container_tests.sh

```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+set -x
+
+cd /project
+
+python -m venv /tmp/venv
+source /tmp/venv/bin/activate
+
+touch requirements_dev.txt
+pip install -r requirements_dev.txt -e .[dev]
+pip freeze --exclude-editable > requirements_dev.txt
+
+pipdeptree
+
+# ensure non-zero length requirements.txt
+echo "# runtime dependencies" >> requirements.txt
```

Comment:
not sure about this. It feels a bit dirty in the Dockerfile. This non-zero length is a requirement of code.yml because it publishes requirements as release assets and they must not be zero length. code.yml calls this test script and it does the release. You could argue that this should be done in code.yml perhaps?

Reply:
that's rather clever. almost too clever, but I think I prefer it !